pkg/pb/pipeline/pipeline.go

```diff
@@ -23,13 +27,59 @@ func (m *Message) Size() int {
 }
 
 func (m *Message) GetID() uint64 {
-	return m.Sid
+	return m.Id
 }
 
 func (m *Message) SetID(id uint64) {
-	m.Sid = id
+	m.Id = id
 }
 
 func (m *Message) DebugString() string {
-	return fmt.Sprintf("sid: %v, cmd: %v, code: %s", m.Sid, m.Cmd, m.Code)
+	errStr := string(m.Err)
+	if m.IsMoErr {
+		me := moerr.Error{}
+		_ = me.UnmarshalBinary(m.Err)
+		errStr = me.Error()
+	}
+	return fmt.Sprintf("sid: %v, cmd: %v, data: %s, err: %s", m.Sid, m.Cmd, m.Data, errStr)
+}
+
+func (m *Message) IsEndMessage() bool {
+	return m.Sid == MessageEnd
+}
+
+func EncodedMessageError(err error) ([]byte, bool) {
```

Comment:
This code does not belong here.   Yuesheng added moerr serial/deserial code.   
You should not have seen any non mo error code, if you really do, either fix, or call ConvertGoError.

Reply:
I still received non mo error because they call functions of third-part package or golang package, and didn't convert the return error to MO-Error.
If I don't distinguish between error cases here but just convert all error data to be MO-Error, a series of bvt cases will be failed. @fengttt

Reply:
Too bad, whoever got the non mo-error, should convert immediately.   Make a list of those, file bugs and ask people to fix.

Here, at least you can call ConvertGoError.   Nobody should use error type cast outside of moerr.

Reply:
ok, I will remove the errors2 and convert all them to be mo err if I received non mo err.

Reply:
#5270